students/swietczak_monika/lesson_12_debugging/verbose_output.py

```diff
@@ -0,0 +1,57 @@
+import os
+import re
+import argparse
+import logging
+
+logging.basicConfig(format=' %(asctime)s - %(levelname)s- %(message)s')
+
+
+def check_args(args=None):
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-r", "--regex",
+                        help="regex",
+                        default=".*")
+    parser.add_argument("-d", "--directory",
+                        help="directory",
+                        default=r"C:\Python36\files")
+    parser.add_argument("-v", "--verbose",
+                        help="verbose level",
+                        choices=['disabled', 'warning', 'info'],
+                        default="warning")
+    results = parser.parse_args(args)
+    return results.regex, results.directory, results.verbose
+
+
+def set_verbose_level(verbose_level):
+    logger = logging.getLogger()
+    if verbose_level == "info":
+        logger.setLevel(logging.INFO)
+    elif verbose_level == "warning":
+        logger.setLevel(logging.WARNING)
+    else:
+        logger.disabled = True
+
+
+def regex_search(reg, folder):
+    regex_text = re.compile(reg)
+    logging.info("Regex = {}".format(regex_text))
+    files_list = os.listdir(folder)
+    logging.info("Files: {}".format(files_list))
```

Comment:
It's good to use logger that knows where you are. That is `logger = logging.getLogger(__name__)` at the beginning of the file and then `logger.info(blah)`.